cmd/batsExecuteTests.go

```diff
@@ -0,0 +1,109 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	pipergit "github.com/SAP/jenkins-library/pkg/git"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/go-git/go-git/v5"
+	"github.com/pkg/errors"
+)
+
+type batsExecuteTestsUtils interface {
+	CloneRepo(URL string) error
+	Stdin(in io.Reader)
+	Stdout(out io.Writer)
+	Stderr(err io.Writer)
+	SetEnv([]string)
+	FileWrite(path string, content []byte, perm os.FileMode) error
+	RunExecutable(e string, p ...string) error
+}
+
+type batsExecuteTestsUtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+}
+
+func newBatsExecuteTestsUtils() batsExecuteTestsUtils {
+	utils := batsExecuteTestsUtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
+func batsExecuteTests(config batsExecuteTestsOptions, telemetryData *telemetry.CustomData) {
+	utils := newBatsExecuteTestsUtils()
+	err := runBatsExecuteTests(&config, telemetryData, utils)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runBatsExecuteTests(config *batsExecuteTestsOptions, telemetryData *telemetry.CustomData, utils batsExecuteTestsUtils) error {
+	if config.OutputFormat != "tap" && config.OutputFormat != "junit" {
+		log.SetErrorCategory(log.ErrorConfiguration)
+		return fmt.Errorf("output format '%v' is incorrect. Possible drivers: tap, junit", config.OutputFormat)
+	}
+
+	err := utils.CloneRepo(config.Repository)
+
+	if err != nil && !errors.Is(err, git.ErrRepositoryAlreadyExists) {
+		return fmt.Errorf("couldn't pull %s repository: %w", config.Repository, err)
+	}
+
+	tapOutput := bytes.Buffer{}
+	utils.Stdout(io.MultiWriter(&tapOutput, log.Writer()))
+	err = utils.RunExecutable("bats-core/bin/bats", "--recursive", "--tap", config.TestPath)
+	if err != nil {
+		return fmt.Errorf("failed to run bats test: %w", err)
+	}
+
+	err = utils.FileWrite("TEST-"+config.TestPackage+".tap", tapOutput.Bytes(), 0644)
+	if err != nil {
+		return fmt.Errorf("failed to write tap file: %w", err)
+	}
+
+	if config.OutputFormat == "junit" {
```

Comment:
container definition required in `batsExecuteTersts.yaml` in order to make sure that everything runs inside a container in case `junit` is used.

Not sure if this has some side-effects though.